Src/Codurance.Domain/GameOfLife.cs

```diff
@@ -0,0 +1,223 @@
+﻿using System;
+using System.Diagnostics;
+using System.Text;
+using static GameOfLife.Domain.ErrorMessages;
+
+namespace GameOfLife.Domain
+{
+    public class GameOfLife : IGameOfLife
+    {
+        public const uint MINIMUM_MATRIX_SIZE = 2;
+        private event EventHandler CalculateLifeExpectancies;
+        private event EventHandler TransferLifeStates;
+
+        public GameOfLife(uint matrixSize)
+        {
+            if (matrixSize < MINIMUM_MATRIX_SIZE)
+            {
+                throw new ArgumentOutOfRangeException(MatrixSizeOutOfRange);
```

Comment:
Should not throw exceptions on a constructor, consider moving the logic onto the generate method